### DISCUSSION TITLE
I cannot reproduce what your UI is doing.

### DISCUSSION THREAD

**Discussion by ivanlengyel** [rom1504/clip-retrieval#104] [Q&A]
Hi. I am trying to search over the LAION dataset using custom text.

Basically I want to reproduce what you are doing with your UI.
I downloaded the files from here: https://deploy.laion.ai/8f83b608504d46bb81708ec86e912220/

```python
# embed text
device = 'cuda'
clip_model="ViT-B/32"
model, preprocess = clip.load(clip_model, device=device, jit=False)

text = "a dog on the snow"
text_tokens = clip.tokenize([text], truncate=True)
text_features = model.encode_text(text_tokens.to(device))
text_features /= text_features.norm(dim=-1, keepdim=True)
text_embeddings = text_features.cpu().detach().numpy().astype('float32')

# find embedding
ind = faiss.read_index("/datasets/indexs_32/text.index")
D, I = ind.search(text_embeddings, 10000)
```
the indexes provided by the search do not match the ones in the parquet files. And those who match the parquet files do not have nothing to do with the text query.
I was hoping that I was going to be able to reproduce and obtain similar images to the ones in [your UI.](https://rom1504.github.io/clip-retrieval/?back=https%3A%2F%2Fknn.laion.ai&index=laion_400m_128G&useMclip=false&query=a+dog+on+the+snow)
Any hint?

I think that it could be some kind of index mapping in the file `ivf_old_to_new_mapping.npy` but I am not able to open it.

Thank you.

**Comment by rom1504** [comment]
https://deploy.laion.ai/8f83b608504d46bb81708ec86e912220/embeddings/metadata/ did you use these parquet files sorted in alphabetical order ?

**Reply by MasterScrat** [reply]
Just to confirm: when you say "alphabetical", you do mean the first 5 files should be:

```
metadata_0.parquet
metadata_1.parquet
metadata_10.parquet
metadata_100.parquet
metadata_101.parquet 
```

as opposed to this, right?

```
metadata_0.parquet
metadata_1.parquet
metadata_2.parquet
metadata_3.parquet
metadata_4.parquet 
```

**Reply by rom1504** [reply]
Yes

**Comment by ivanlengyel** [comment]
I am using [these ones](https://deploy.laion.ai/8f83b608504d46bb81708ec86e912220/dataset/).
Which dataframe indexes did you use to construct the faiss index?

I guess that not the ones that I am using...

**Reply by rom1504** [reply]
Yes that's the problem, you need to use the ones of the link I sent above

**Reply by ivanlengyel** [reply]
Ok, thank you!

**Comment by rom1504** [comment]
Yes
